src/components/PriceChart/PriceChart.js

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import axios from "axios";
+import Chart from "chart.js/auto";
+import { Line } from "react-chartjs-2";
+import { getPreviousDates, formatOverviewNumber, startDate, wordedDate } from "utils";
+import { StyledPriceChart } from './PriceChart.styles';
+
+export class PriceChart extends React.Component {
+    state = {
+        isLoading: false,
+        error: null,
+        priceData: [],
+        todaysPrice: null,
+    }
+
+    getMarketChartData = async () => {
+        try {
+            this.setState({ isLoading: true });
+            const { currency, timeSpan } = this.props
+            let span;
+            switch (timeSpan) {
+                case "1":
+                    span = 30
+                    break;
+                case "3":
+                    span = 90
+                    break;
+                case "6":
+                    span = 180
+                    break;
+                case "12":
+                    span = 360
+                    break;
+                default: span = 7
+                    break;
+            }
+
+            const response = await axios(
+                `https://api.coingecko.com/api/v3/coins/bitcoin/market_chart?vs_currency=${currency}&days=${span}&interval=daily`
+            );
+            const priceData = response.data.prices.map((price) => price[1]).slice(1, (span + 1));
+            const todaysPrice = response.data.prices[0][1]
+            this.setState({ isLoading: false, priceData: priceData, todaysPrice: todaysPrice });
+        } 
+        catch (err) {
+            console.error(err)
+            this.setState({ isLoading: false, error: err });
+        }
+    };
+
+    componentDidMount() {
+        this.getMarketChartData()
+    }
+    componentDidUpdate(prevProps) {
+        if(prevProps.currency !== this.props.currency || prevProps.timeSpan !== this.props.timeSpan) {
+            this.getMarketChartData()
+        }
```

Comment:
if(prevProps.currency !== this.props.currency) {
            this.getMarketChartData()
}

 if(prevProps.timeSpan !== this.props.timeSpan) {
            this.getMarketChartData()
}